src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -337,267 +214,210 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
-                    if (packageContent != null)
-                    {
-                        metadata.VersionDownloadUri = packageContent.ToString();
-                    }
-                    
-                    // size and hash from versionContent
-                    metadata.Size = OssUtilities.GetJSONPropertyIfExists(versionContent, "packageSize")?.GetInt64();
-                    metadata.Signature ??= new List<Digest>();
-                    metadata.Signature.Add(new Digest
-                    {
-                        Algorithm = OssUtilities.GetJSONPropertyStringIfExists(versionContent, "packageHashAlgorithm"),
-                        Signature = OssUtilities.GetJSONPropertyStringIfExists(versionContent, "packageHash"),
-                    });
-                    
-                    // homepage
-                    metadata.Homepage = OssUtilities.GetJSONPropertyStringIfExists(versionContent, "projectUrl");
-
-                    // author(s)
-                    JsonElement? authorElement = OssUtilities.GetJSONPropertyIfExists(versionElement, "authors");
-                    User author = new();
-                    if (authorElement is not null)
-                    {
-                        author.Name = authorElement?.GetString();
-                        // TODO: User email and url
-                        // author.Email = OssUtilities.GetJSONPropertyStringIfExists(authorElement, "email");
-                        // author.Url = OssUtilities.GetJSONPropertyStringIfExists(authorElement, "url");
+            return metadata;
+        }
 
-                        metadata.Authors ??= new List<User>();
-                        metadata.Authors.Add(author);
-                    }
+        /// <summary>
+        /// Updates the package version specific values in <see cref="PackageMetadata"/>.
+        /// </summary>
+        /// <param name="metadata">The <see cref="PackageMetadata"/> object to update with the values for this version.</param>
+        /// <param name="packageVersion">The <see cref="NuGetMetadata"/> representing this version.</param>
+        private async Task UpdateVersionMetadata(PackageMetadata metadata, NuGetMetadata packageVersion)
+        {
+            if (metadata.PackageVersion is null)
+            {
+                return;
+            }
 
-                    // TODO: maintainers
+            string? nameLowercase = metadata.Name?.ToLowerInvariant();
 
-                    // repository
-                    PackageURL nuspecPurl = purl;
+            // Set the version specific URI values.
+            metadata.VersionUri = $"{metadata.PackageManagerUri}/packages/{nameLowercase}/{metadata.PackageVersion}";
+            metadata.ApiVersionUri = packageVersion.CatalogUri.ToString();
+            
+            // Construct the artifact contents url.
+            metadata.VersionDownloadUri = GetNupkgUrl(metadata.Name!, metadata.PackageVersion);
```

Comment:
The fact that we are using `metadata.Name!` here also suggests that we should be actually checking whether `metadata.Name` is null or not earlier (see my above comment)

Reply:
Resolved in other PR.